src/swap/sideshift.js

```diff
@@ -0,0 +1,310 @@
+// @flow
+
+import { gt, lt, mul } from 'biggystring'
+import {
+  SwapAboveLimitError,
+  SwapBelowLimitError,
+  SwapPermissionError
+} from 'edge-core-js'
+import {
+  type EdgeCorePluginOptions,
+  type EdgeCurrencyWallet,
+  type EdgeFetchResponse,
+  type EdgeSpendInfo,
+  type EdgeSwapInfo,
+  type EdgeSwapPlugin,
+  type EdgeSwapQuote,
+  type EdgeSwapRequest,
+  type EdgeTransaction,
+  SwapCurrencyError
+} from 'edge-core-js/types'
+
+import { makeSwapPluginQuote } from '../swap-helpers.js'
+
+// Invalid currency codes should *not* have transcribed codes
+// because currency codes with transcribed versions are NOT invalid
+const CURRENCY_CODE_TRANSCRIPTION = {
+  // Edge currencyCode: exchangeCurrencyCode
+  USDT: 'usdtErc20'
+}
+
+const pluginId = 'sideshift'
+const swapInfo: EdgeSwapInfo = {
+  pluginId,
+  displayName: 'SideShift.ai',
+  supportEmail: 'help@sideshift.ai'
+}
+type FixedQuote = {
+  createdAt: string,
+  depositAmount: string,
+  depositMethod: string,
+  expiresAt: string,
+  id: string,
+  rate: string,
+  settleAmount: string,
+  settleMethod: string
+}
+
+type FixedQuoteRequestParams = {
+  depositMethod: string,
+  settleMethod: string,
+  depositAmount: string
+}
+
+type OrderRequest = {
+  createdAt: string,
+  createdAtISO: string,
+  expiresAt: string,
+  expiresAtISO: string,
+  depositAddress: {
+    address: string
+  },
+  depositMethod: string,
+  id: string,
+  orderId: string,
+  settleAddress: {
+    address: string
+  },
+  settleMethod: string,
+  depositMax: string,
+  depositMin: string,
+  quoteId: string,
+  settleAmount: string,
+  depositAmount: string,
+  deposits: Array<any>
+}
+
+type OrderRequestParams = {
+  type: string,
+  quoteId: string,
+  affiliateId: string,
+  sessionSecret?: string,
+  settleAddress: string
+}
+
+type Rate = {
+  rate: number,
+  min: string,
+  max: string,
+  error?: {
+    message: string
+  }
+}
+
+type Permission = {
+  createOrder: boolean,
+  createQuote: boolean
+}
+
+const dontUseLegacy = {}
+
+async function getAddress(
+  wallet: EdgeCurrencyWallet,
+  currencyCode: string
+): Promise<string> {
+  const addressInfo = await wallet.getReceiveAddress({ currencyCode })
+  return addressInfo.legacyAddress && !dontUseLegacy[currencyCode]
+    ? addressInfo.legacyAddress
+    : addressInfo.publicAddress
+}
+
+async function checkReplyForError(reply: EdgeFetchResponse): Promise<any> {
```

Comment:
@nilswx I suggest a `function createSideShiftApi(baseUrl: string)` that returns the `get` and `post` methods inside an object. This way we can get rid of the oddly placed `checkReplyForError` and reduce code inside plugin